test/functional/feature_rbf.py

```diff
@@ -129,33 +128,27 @@ def run_test(self):
 
     def test_simple_doublespend(self):
         """Simple doublespend"""
-        tx0_outpoint = make_utxo(self.nodes[0], int(1.1 * COIN))
+        # we use MiniWallet to create a transaction template with inputs correctly set,
+        # and modify the output (amount, scriptPubKey) according to our needs
+        utxo = self.wallet.get_utxo()
+        tx_template = self.wallet.create_self_transfer(from_node=self.nodes[0], utxo_to_spend=utxo)['tx']
 
-        # make_utxo may have generated a bunch of blocks, so we need to sync
-        # before we can spend the coins generated, or else the resulting
-        # transactions might not be accepted by our peers.
-        self.sync_all()
-
-        tx1a = CTransaction()
-        tx1a.vin = [CTxIn(tx0_outpoint, nSequence=0)]
+        tx1a = deepcopy(tx_template)
         tx1a.vout = [CTxOut(1 * COIN, DUMMY_P2WPKH_SCRIPT)]
         tx1a_hex = tx1a.serialize().hex()
         tx1a_txid = self.nodes[0].sendrawtransaction(tx1a_hex, 0)
 
         self.sync_all()
```

Comment:
I'm not not sure why this call to `self.sync_all()` is here. Looks like you removed it on line 137 and it's also not used after calling `sendrawtransaction` on line 155. Since the miniwallet is attached to node 0 and its the only once we are referencing in the test, it seems unnecessary.

I deleted line 141 while testing, ran the test a few times and every run passed. I'd suggest removing it.

Reply:
Thanks, excellent find! Removed the sync_all() call in the latest force-push.